core/src/elements/ons-ripple/index.js

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2013-2015 ASIAL CORPORATION
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import util from 'ons/util';
+import BaseElement from 'ons/base-element';
+import GestureDetector from 'ons/gesture-detector';
+import AnimatorJS from './animator-js';
+import AnimatorCSS from './animator-css';
+
+var Animator = [AnimatorJS, AnimatorCSS][1];
+
+class RippleElement extends BaseElement {
+
+  createdCallback() {
+    this.classList.add('ripple');
+    if (this.hasAttribute('_compiled')) {
+      ['_wave', '_background'].forEach(e => {
+        this[e] = this.getElementsByClassName('ripple_' + e)[0];
+      });
+    } else {
+      this._compile();
+    }
+
+    this._animator = new Animator();
+    this._gestureDetector = new GestureDetector(this, {holdTimeout: 300});
+
+    ['color', 'target', 'center', 'start-radius', 'background'].forEach(e => {
+      this.attributeChangedCallback(e, null, this.getAttribute(e));
+    });
+  }
+
+
+  _compile() {
+    ['_wave', '_background'].forEach(e => {
+      this[e] = document.createElement('div');
+      this[e].classList.add('ripple_' + e);
+      this.appendChild(this[e]);
```

Comment:
Document fragments increase performance mostly because they make all changes to the DOM at once, hence making the browser do only one reflow instead of multiple ones. 

However if the modifications are made outside the DOM, using document fragment actually slows down the process - [jsperf test 1](http://jsperf.com/createdocumentfragment-vs-appendchild-on-non-appended-e/32) . In our web components, as far as I understand - _compile is executed in createdCallback, not in attachedCallback, so the actions here are before the element is attached to the DOM.

In this case also even if it was inside the DOM I am appending only 2 children - again at least in the current browsers creating a fragment for 2 children does not seem to be beneficial, and again makes it slower - [jsperf test 2](http://jsperf.com/docfragment-test/14).

Therefore I have not modified this part in the recent commit.

In the second jsperf I have also added another case if the code for the two elements is copy-pasted. There seems to be around 3% performance improvement if its done. There are probably also other places where I could make the same change if we want to squeeze out the best performance. If you think this is necessary please tell me and I will fix them.